packages/webdriverio/src/commands/element/addValue.js

```diff
@@ -28,7 +28,7 @@
 import { transformToCharString } from '../../utils'
 
 export default function addValue (value) {
-    if (!this.isW3C) {
+    if (!this.isW3C || this.isMobileBrowser) {
```

Comment:
I don't think we need to introduce `isMobileBrowser` here. Just `isMobile` seems enough, no?
```suggestion
    if (!this.isW3C || this.isMobile) {
```

Reply:
I didn't notice the error occurring for any o the app tests.  I assumed the protocol used is different for browser/app testing in mobile.

Reply:
@christian-bromann Would you be able to help me out with this problem?

Reply:
The problem with this is that the WebDriver endpoint shouldn't be different between desktop browser and mobile browser because both are run by the same driver. Therefor introducing a `isMobileBrowser` seems unnecessary. On which environment are you running the tests? If it is Appium on mobile Chrome can you extract the chromedriver logs?

Reply:
@christian-bromann I agree with you. We should see the same results running the same test against the mobile chrome and desktop chrome. But in here the test passes in desktop chrome and fails in mobile chrome. I can pull out the chromedriver log. The other difference I can think of is the service we are using is different. In mobile I use Appium and browser it is selenium. Another thing to node I tested this with v4, everything looks good.

Reply:
```
[HTTP] {"using":"css selector","value":"#search_input_react"}
[W3C] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Matched '/wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element' to command name 'findElement'
[JSONWP Proxy] Proxying [POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element] to [POST http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9/element] with body: {"using":"css selector","value":"#search_input_react"}
[Chromedriver] [STDERR] [1547747678.881][INFO]: COMMAND FindElement {
[Chromedriver] [STDERR]    "using": "css selector",
[Chromedriver] [STDERR]    "value": "#search_input_react"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.881][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747678.881][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=27) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.915][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=27) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.915][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747678.917][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=28) {
[Chromedriver] [STDERR]    "expression": "(function() { // Copyright (c) 2012 The Chromium Authors. All rights reserved.\n// Use of this source code is governed by a BSD-style license that can be\n// found in the LICENSE file.\n\n/**\n * Enum f...",
[Chromedriver] [STDERR]    "returnByValue": true
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.959][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=28) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "type": "object",
[Chromedriver] [STDERR]       "value": {
[Chromedriver] [STDERR]          "status": 0,
[Chromedriver] [STDERR]          "value": {
[Chromedriver] [STDERR]             "ELEMENT": "0.733588335143198-1"
[Chromedriver] [STDERR]          }
[Chromedriver] [STDERR]       }
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.959][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747678.959][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=29) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.964][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=29) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747678.964][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747678.964][INFO]: RESPONSE FindElement {
[Chromedriver] [STDERR]    "ELEMENT": "0.733588335143198-1"
[Chromedriver] [STDERR] }
[JSONWP Proxy] Got response with status 200: {"sessionId":"18ab5510595cb190e27d7247699a92f9","status":0,"value":{"ELEMENT":"0.733588335143198-1"}}
[JSONWP Proxy] Replacing sessionId 18ab5510595cb190e27d7247699a92f9 with df83aea9-757e-466b-8a1b-962065cc8054
[HTTP] <-- POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element 200 89 ms - 105
[HTTP] 
[HTTP] --> POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/clear
[HTTP] {}
[W3C] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Matched '/wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/clear' to command name 'clear'
[JSONWP Proxy] Proxying [POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/clear] to [POST http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9/element/0.733588335143198-1/clear] with body: {}
[Chromedriver] [STDERR] [1547747679.024][INFO]: COMMAND ClearElement {
[Chromedriver] [STDERR]    "id": "0.733588335143198-1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.024][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.024][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=30) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.029][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=30) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.029][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.030][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=31) {
[Chromedriver] [STDERR]    "expression": "(function() { // Copyright (c) 2012 The Chromium Authors. All rights reserved.\n// Use of this source code is governed by a BSD-style license that can be\n// found in the LICENSE file.\n\n/**\n * Enum f...",
[Chromedriver] [STDERR]    "returnByValue": true
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.083][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=31) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "type": "object",
[Chromedriver] [STDERR]       "value": {
[Chromedriver] [STDERR]          "status": 0
[Chromedriver] [STDERR]       }
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.084][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.084][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=32) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.088][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=32) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.088][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.088][INFO]: RESPONSE ClearElement
[JSONWP Proxy] Got response with status 200: {"sessionId":"18ab5510595cb190e27d7247699a92f9","status":0,"value":null}
[JSONWP Proxy] Replacing sessionId 18ab5510595cb190e27d7247699a92f9 with df83aea9-757e-466b-8a1b-962065cc8054
[HTTP] <-- POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/clear 200 67 ms - 76
[HTTP] 
[HTTP] --> POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value
[HTTP] {"text":"foo"}
[W3C] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Matched '/wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value' to command name 'setValue'
[JSONWP Proxy] Proxying [POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value] to [POST http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9/element/0.733588335143198-1/value] with body: {"text":"foo"}
[Chromedriver] [STDERR] [1547747679.098][INFO]: COMMAND TypeElement {
[Chromedriver] [STDERR]    "id": "0.733588335143198-1",
[Chromedriver] [STDERR]    "text": "foo"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.098][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.098][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=33) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.114][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=33) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.114][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.114][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=34) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.128][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=34) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.128][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.128][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=35) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.137][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=35) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.137][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.137][INFO]: RESPONSE TypeElement unknown error: 'value' must be a list
[Chromedriver] [STDERR]   (Session info: chrome=71.0.3578.99)
[JSONWP Proxy] Got response with status 200: {"sessionId":"18ab5510595cb190e27d7247699a92f9","status":13,"value":{"message":"unknown error: 'value' must be a list\n  (Session info: chrome=71.0.3578.99)\n  (Driver info: chromedriver=2.40.565498 (ea082db3280dd6843ebfb08a625e3eb905c4f5ab),platform=Windows NT 10.0.16299 x86_64)"}}
[W3C] Encountered internal error running command: ProxyRequestError: Could not proxy command to remote server. Original error: The request to /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value has failed
[W3C]     at JWProxy.proxy$ (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\jsonwp-proxy\proxy.js:182:13)
[W3C]     at tryCatch (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
[W3C]     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
[W3C]     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:100:21)
[W3C]     at GeneratorFunctionPrototype.invoke (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
[W3C]     at 
[MJSONWP] Matched JSONWP error code 13 to UnknownError
[HTTP] <-- POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value 500 84 ms - 2080
[HTTP] 
[HTTP] --> POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value
[HTTP] {"text":"foo"}
[W3C] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Matched '/wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value' to command name 'setValue'
[JSONWP Proxy] Proxying [POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value] to [POST http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9/element/0.733588335143198-1/value] with body: {"text":"foo"}
[Chromedriver] [STDERR] [1547747679.186][INFO]: COMMAND TypeElement {
[Chromedriver] [STDERR]    "id": "0.733588335143198-1",
[Chromedriver] [STDERR]    "text": "foo"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.186][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.186][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=36) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.316][DEBUG]: DEVTOOLS EVENT Log.entryAdded {
[Chromedriver] [STDERR]    "entry": {
[Chromedriver] [STDERR]       "level": "warning",
[Chromedriver] [STDERR]       "lineNumber": 244,
[Chromedriver] [STDERR]       "source": "javascript",
[Chromedriver] [STDERR]       "stackTrace": {
[Chromedriver] [STDERR]          "callFrames": [ {
[Chromedriver] [STDERR]             "columnNumber": 21,
[Chromedriver] [STDERR]             "functionName": "de",
[Chromedriver] [STDERR]             "lineNumber": 244,
[Chromedriver] [STDERR]             "scriptId": "53",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/player-vfl5cSuiu/en_US/base.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 408,
[Chromedriver] [STDERR]             "functionName": "f0",
[Chromedriver] [STDERR]             "lineNumber": 4487,
[Chromedriver] [STDERR]             "scriptId": "53",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/player-vfl5cSuiu/en_US/base.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 389,
[Chromedriver] [STDERR]             "functionName": "Exa",
[Chromedriver] [STDERR]             "lineNumber": 4509,
[Chromedriver] [STDERR]             "scriptId": "53",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/player-vfl5cSuiu/en_US/base.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 385,
[Chromedriver] [STDERR]             "functionName": "b0",
[Chromedriver] [STDERR]             "lineNumber": 4482,
[Chromedriver] [STDERR]             "scriptId": "53",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/player-vfl5cSuiu/en_US/base.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 160,
[Chromedriver] [STDERR]             "functionName": "",
[Chromedriver] [STDERR]             "lineNumber": 7996,
[Chromedriver] [STDERR]             "scriptId": "53",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/player-vfl5cSuiu/en_US/base.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 260,
[Chromedriver] [STDERR]             "functionName": "a.u",
[Chromedriver] [STDERR]             "lineNumber": 510,
[Chromedriver] [STDERR]             "scriptId": "50",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/www-embed-player-vflYYxnqe/www-embed-player.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 11,
[Chromedriver] [STDERR]             "functionName": "ch",
[Chromedriver] [STDERR]             "lineNumber": 511,
[Chromedriver] [STDERR]             "scriptId": "50",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/www-embed-player-vflYYxnqe/www-embed-player.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 3,
[Chromedriver] [STDERR]             "functionName": "h.ca",
[Chromedriver] [STDERR]             "lineNumber": 506,
[Chromedriver] [STDERR]             "scriptId": "50",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/www-embed-player-vflYYxnqe/www-embed-player.js"
[Chromedriver] [STDERR]          }, {
[Chromedriver] [STDERR]             "columnNumber": 243,
[Chromedriver] [STDERR]             "functionName": "",
[Chromedriver] [STDERR]             "lineNumber": 534,
[Chromedriver] [STDERR]             "scriptId": "50",
[Chromedriver] [STDERR]             "url": "https://www.youtube.com/yts/jsbin/www-embed-player-vflYYxnqe/www-embed-player.js"
[Chromedriver] [STDERR]          } ]
[Chromedriver] [STDERR]       },
[Chromedriver] [STDERR]       "text": "The deviceorientation events are blocked by feature policy. See https://github.com/WICG/feature-policy/blob/master/features.md#sensor-features",
[Chromedriver] [STDERR]       "timestamp": 1547747679109.93,
[Chromedriver] [STDERR]       "url": "https://www.youtube.com/yts/jsbin/player-vfl5cSuiu/en_US/base.js"
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.522][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=36) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.522][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.522][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=37) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.545][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=37) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.545][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.545][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=38) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.557][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=38) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.557][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.557][INFO]: RESPONSE TypeElement unknown error: 'value' must be a list
[Chromedriver] [STDERR]   (Session info: chrome=71.0.3578.99)
[JSONWP Proxy] Got response with status 200: {"sessionId":"18ab5510595cb190e27d7247699a92f9","status":13,"value":{"message":"unknown error: 'value' must be a list\n  (Session info: chrome=71.0.3578.99)\n  (Driver info: chromedriver=2.40.565498 (ea082db3280dd6843ebfb08a625e3eb905c4f5ab),platform=Windows NT 10.0.16299 x86_64)"}}
[W3C] Encountered internal error running command: ProxyRequestError: Could not proxy command to remote server. Original error: The request to /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value has failed
[W3C]     at JWProxy.proxy$ (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\jsonwp-proxy\proxy.js:182:13)
[W3C]     at tryCatch (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
[W3C]     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
[W3C]     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:100:21)
[W3C]     at GeneratorFunctionPrototype.invoke (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
[W3C]     at 
[MJSONWP] Matched JSONWP error code 13 to UnknownError
[HTTP] <-- POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value 500 375 ms - 2080
[HTTP] 
[HTTP] --> POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value
[HTTP] {"text":"foo"}
[W3C] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Matched '/wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value' to command name 'setValue'
[JSONWP Proxy] Proxying [POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value] to [POST http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9/element/0.733588335143198-1/value] with body: {"text":"foo"}
[Chromedriver] [STDERR] [1547747679.563][INFO]: COMMAND TypeElement {
[Chromedriver] [STDERR]    "id": "0.733588335143198-1",
[Chromedriver] [STDERR]    "text": "foo"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.563][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.563][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=39) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.569][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=39) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.569][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.569][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=40) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.574][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=40) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.574][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.574][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=41) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.577][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=41) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.577][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.577][INFO]: RESPONSE TypeElement unknown error: 'value' must be a list
[Chromedriver] [STDERR]   (Session info: chrome=71.0.3578.99)
[JSONWP Proxy] Got response with status 200: {"sessionId":"18ab5510595cb190e27d7247699a92f9","status":13,"value":{"message":"unknown error: 'value' must be a list\n  (Session info: chrome=71.0.3578.99)\n  (Driver info: chromedriver=2.40.565498 (ea082db3280dd6843ebfb08a625e3eb905c4f5ab),platform=Windows NT 10.0.16299 x86_64)"}}
[W3C] Encountered internal error running command: ProxyRequestError: Could not proxy command to remote server. Original error: The request to /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value has failed
[W3C]     at JWProxy.proxy$ (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\jsonwp-proxy\proxy.js:182:13)
[W3C]     at tryCatch (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
[W3C]     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
[W3C]     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:100:21)
[W3C]     at GeneratorFunctionPrototype.invoke (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
[W3C]     at 
[MJSONWP] Matched JSONWP error code 13 to UnknownError
[HTTP] <-- POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value 500 19 ms - 2080
[HTTP] 
[HTTP] --> POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value
[HTTP] {"text":"foo"}
[W3C] Driver proxy active, passing request on via HTTP proxy
[JSONWP Proxy] Matched '/wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value' to command name 'setValue'
[JSONWP Proxy] Proxying [POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value] to [POST http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9/element/0.733588335143198-1/value] with body: {"text":"foo"}
[Chromedriver] [STDERR] [1547747679.584][INFO]: COMMAND TypeElement {
[Chromedriver] [STDERR]    "id": "0.733588335143198-1",
[Chromedriver] [STDERR]    "text": "foo"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.584][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.584][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=42) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.589][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=42) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.589][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.589][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=43) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.593][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=43) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.593][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547747679.593][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=44) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.597][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=44) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.597][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547747679.597][INFO]: RESPONSE TypeElement unknown error: 'value' must be a list
[Chromedriver] [STDERR]   (Session info: chrome=71.0.3578.99)
[JSONWP Proxy] Got response with status 200: {"sessionId":"18ab5510595cb190e27d7247699a92f9","status":13,"value":{"message":"unknown error: 'value' must be a list\n  (Session info: chrome=71.0.3578.99)\n  (Driver info: chromedriver=2.40.565498 (ea082db3280dd6843ebfb08a625e3eb905c4f5ab),platform=Windows NT 10.0.16299 x86_64)"}}
[W3C] Encountered internal error running command: ProxyRequestError: Could not proxy command to remote server. Original error: The request to /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value has failed
[W3C]     at JWProxy.proxy$ (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\jsonwp-proxy\proxy.js:182:13)
[W3C]     at tryCatch (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:67:40)
[W3C]     at GeneratorFunctionPrototype.invoke [as _invoke] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:315:22)
[W3C]     at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:100:21)
[W3C]     at GeneratorFunctionPrototype.invoke (C:\Users\nvarthakavi\AppData\Local\Programs\Appium\resources\app\node_modules\appium\node_modules\babel-runtime\regenerator\runtime.js:136:37)
[W3C]     at 
[MJSONWP] Matched JSONWP error code 13 to UnknownError
[HTTP] <-- POST /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054/element/0.733588335143198-1/value 500 17 ms - 2080
[HTTP] 
[HTTP] --> DELETE /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054
[HTTP] {}
[W3C] Calling AppiumDriver.deleteSession() with args: ["df83aea9-757e-466b-8a1b-962065cc8054"]
[BaseDriver] Event 'quitSessionRequested' logged at 1547747679634 (09:54:39 GMT-0800 (Pacific Standard Time))
[Appium] Removing session df83aea9-757e-466b-8a1b-962065cc8054 from our master session list
[UiAutomator2] Deleting UiAutomator2 session
[AndroidDriver] Stopping chromedriver for context CHROMIUM
[Chromedriver] Changed state to 'stopping'
[JSONWP Proxy] Proxying [DELETE /] to [DELETE http://127.0.0.1:8000/wd/hub/session/18ab5510595cb190e27d7247699a92f9] with no body
[Chromedriver] [STDERR] [1547747679.634][INFO]: COMMAND Quit {
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547747679.634][DEBUG]: Sending adb command: host:transport:emulator-5554|shell:am force-stop com.android.chrome
[Chromedriver] [STDERR] [1547747679.879][DEBUG]: Received adb response: 
[Chromedriver] [STDERR] [1547747679.879][INFO]: RESPONSE Quit
[Chromedriver] [STDERR] [1547747679.879][DEBUG]: Log type 'driver' lost 0 entries on destruction
[Chromedriver] [STDERR] [1547747679.879][DEBUG]: Log type 'browser' lost 1 entries on destruction
[JSONWP Proxy] Got response with status 200: "{\"sessionId\":\"18ab5510595cb190e27d7247699a92f9\",\"status\":0,\"value\":null}"
[Chromedriver] Changed state to 'stopped'
[UiAutomator2] Deleting UiAutomator2 server session
[JSONWP Proxy] Matched '/' to command name 'deleteSession'
[JSONWP Proxy] Proxying [DELETE /] to [DELETE http://localhost:8200/wd/hub/session/28f0e937-cc9f-4b9e-a401-6e434fed6552] with no body
[JSONWP Proxy] Got response with status 200: "{\"sessionId\":\"28f0e937-cc9f-4b9e-a401-6e434fed6552\",\"status\":0,\"value\":\"Session deleted\"}"
[ADB] Running 'C:\Users\nvarthakavi\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 shell am force-stop com.android.chrome'
[Logcat] Stopping logcat capture
[ADB] Removing forwarded port socket connection: 8200 
[ADB] Running 'C:\Users\nvarthakavi\AppData\Local\Android\Sdk\platform-tools\adb.exe -P 5037 -s emulator-5554 forward --remove tcp\:8200'
[BaseDriver] Event 'quitSessionFinished' logged at 1547747680502 (09:54:40 GMT-0800 (Pacific Standard Time))
[W3C] Received response: null
[W3C] But deleting session, so not returning
[W3C] Responding to client with driver.deleteSession() result: null
[HTTP] <-- DELETE /wd/hub/session/df83aea9-757e-466b-8a1b-962065cc8054 200 869 ms - 14
[HTTP] 
[UiAutomator2] [UIAutomator2] io.appium.uiautomator2.server.test.AppiumUiAutomator2Server:.
[UiAutomator2] [UIAutomator2] Time: 18.074
[UiAutomator2] [UIAutomator2] OK (1 test)
```
 Attaching the Logs

Reply:
```
[10:13:29]  COMMAND     POST     "/wd/hub/session/e4d971bc-25db-469f-ad22-60043ef71257/element/0.19691049715603692-1/value"
[10:13:29]  DATA                {"value":["f","o","o"],"text":"foo"}
[Chromedriver] [STDERR] [1547748809.058][INFO]: Done waiting for pending navigations. Status: ok
[Chromedriver] [STDERR] [1547748809.058][INFO]: RESPONSE ClearElement
[HTTP] --> POST /wd/hub/session/e4d971bc-25db-469f-ad22-60043ef71257/element/0.19691049715603692-1/value
[HTTP] {"value":["f","o","o"],"text":"foo"}
[MJSONWP] Driver proxy active, passing request on via HTTP proxy
[debug] [JSONWP Proxy] Matched '/wd/hub/session/e4d971bc-25db-469f-ad22-60043ef71257/element/0.19691049715603692-1/value' to command name 'setValue'
[debug] [JSONWP Proxy] Proxying [POST /wd/hub/session/e4d971bc-25db-469f-ad22-60043ef71257/element/0.19691049715603692-1/value] to [POST http://127.0.0.1:8000/wd/hub/session/18a6c4c6d2d60094ca003f7b85ff7ae7/element/0.19691049715603692-1/value] with body: {"value":["f","o","o"],"text":"foo"}
[Chromedriver] [STDERR] [1547748809.075][INFO]: COMMAND TypeElement {
[Chromedriver] [STDERR]    "id": "0.19691049715603692-1",
[Chromedriver] [STDERR]    "text": "foo",
[Chromedriver] [STDERR]    "value": [ "f", "o", "o" ]
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547748809.075][INFO]: Waiting for pending navigations...
[Chromedriver] [STDERR] [1547748809.075][DEBUG]: DEVTOOLS COMMAND Runtime.evaluate (id=36) {
[Chromedriver] [STDERR]    "expression": "1"
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547748809.086][DEBUG]: DEVTOOLS RESPONSE Runtime.evaluate (id=36) {
[Chromedriver] [STDERR]    "result": {
[Chromedriver] [STDERR]       "description": "1",
[Chromedriver] [STDERR]       "type": "number",
[Chromedriver] [STDERR]       "value": 1
[Chromedriver] [STDERR]    }
[Chromedriver] [STDERR] }
[Chromedriver] [STDERR] [1547748809.086][INFO]: Done waiting for pending navigations. Status: ok
``` 

These logs from v4 tests.

Reply:
If you just look at the command for `setValue()` the passing value is different for both cases.

Reply:
@christian-bromann I seriously cannot explain this discrepency. In v4 the transformtoCharString is applied to every type of command. and in V5 only for commands that are not `W3C`

Reply:
I have also updated the changes that you requested.